tests/test_lowlevel.py

```diff
@@ -25,6 +25,71 @@ def test_chunked_upload():
     assert r.request.headers['Transfer-Encoding'] == 'chunked'
 
 
+def test_digestauth_401_count_reset_on_redirect():
+    """Ensure we correctly reset num_401_calls after a successful digest auth,
+    followed by a 302 redirect to another digest auth prompt.
+
+    See https://github.com/kennethreitz/requests/issues/1979.
+    """
+    text_401 = ('HTTP/1.1 401 UNAUTHORIZED\r\n'
+                'Connection:keep-alive\r\n'
+                'Content-Length: 0\r\n'
+                'Set-Cookie:fake=fake_value\r\n'
+                'WWW-Authenticate:Digest nonce="6bf5d6e4da1ce66918800195d6b9130d"'
```

Comment:
That's a good question, I copied these directly from an httpbin response to Requests. I'll separate them out.